odp_sdk/resource_client.py

```diff
@@ -0,0 +1,216 @@
+        try:
+            res.raise_for_status()
+        except requests.HTTPError as e:
+            if res.status_code == 401:
+                raise OdpUnauthorizedError("Unauthorized request") from e
+            if res.status_code == 404:
+                raise OdpValidationError("API argument error") from e
+            raise  # Unhandled error
+
+        content = res.json()
+        return [ResourceDto(**item) for item in content["results"]], content.get("next")
+
+    def create(self, manifest: ResourceDto) -> ResourceDto:
+        """Create a resource from a manifest
+
+        Args:
+            manifest: Resource manifest
+
+        Returns:
+            The manifest of the created resource, populated with uuid and status
+
+        Raises:
+            OdpUnauthorizedError: Unauthorized request
+            OdpValidationError: Invalid input
+            OdpResourceExistsError: Resource already exists
+        """
+
+        res = self.http_client.post(self.resource_endpoint, content=manifest.dict())
+        try:
+            res.raise_for_status()
+        except requests.HTTPError as e:
+            if res.status_code == 400:
+                raise OdpValidationError("Invalid input") from e
+            if res.status_code == 401:
+                raise OdpUnauthorizedError("Unauthorized request") from e
+            if res.status_code == 409:
+                raise OdpResourceExistsError("Resource already exists") from e
+            raise  # Unhandled error
+
+        return ResourceDto(**res.json())
+
+    def update(self, manifest_update: ResourceDto | dict, ref: Optional[str | UUID] = None) -> ResourceDto:
+        """Update a resource from a manifest
+
+        Args:
+            manifest_update: Resource manifest or JSON patch
+            ref: Optional reference to the resource to update.
+
+        Returns:
+            The manifest of the updated resource, populated with the updated fields
+
+        Raises:
+            OdpUnauthorizedError: Unauthorized request
+            OdpValidationError: Invalid input
+            OdpResourceNotFoundError: Resource not found
+        """
+        if ref:
+            if isinstance(ref, UUID):
+                params = {"either_id": str(ref)}
+            else:
+                rg, rt, name = ref.split("/")
+                params = {"either_id": name, "kind": f"{rg}/{rt}"}
+        else:
+            params = {}
+
+        res = self.http_client.patch(self.resource_endpoint, params=params, content=manifest_update)
+        try:
+            res.raise_for_status()
+        except requests.HTTPError as e:
+            if res.status_code == 400:
+                raise OdpValidationError("Invalid input") from e
+            if res.status_code == 401:
+                raise OdpUnauthorizedError("Unauthorized request") from e
+            if res.status_code == 404:
+                raise OdpResourceNotFoundError("Resource not found") from e
```

Comment:
I know it's a long stretch but can we also add forbidden(403) as well for some verb like update, delete etc. If its not defined then we implement it in the future.

Reply:
good point. The API currently does not support this. Can you create an issue on this?